charts/opentelemetry-collector/values.yaml

```diff
@@ -110,25 +110,21 @@ ports:
     enabled: true
     containerPort: 4317
     servicePort: 4317
-    hostPort: 4317
```

Comment:
Can you please keep the hostPorts enabled conditionally if being run as a daemonset (agent) ?

Reply:
@naseemkullah Could you explain why you would want it by default? 
without using hostPort, it can be accessed through using k8s service. 
in my AWS EKS cluster, it fails to be deployed when trying to use hostPort. it seems that it requires additional cluster configuration to map it to host port. It would make it a bit more complicated to deploy it.

Reply:
This how the instrumented workloads communicate with the collector when it runs as a daemon set in agent mode, I agree that in standalone collector mode they should be disabled and the k8s service used, hence enabling them under the condition that agent mode is enabled.

Reply:
so pods that are running apps that has been instrumented to send trace information are sending it to its node ip it is running on?

Reply:
thats right! with something like this in their env:

```
      - name: OTEL_EXPORTER_JAEGER_AGENT_HOST
        valueFrom:
          fieldRef:
            fieldPath: status.hostIP
 ```

Reply:
the alternative is running the otel-collector agent as a sidecar container, which is outside the scope of this helm chart

Reply:
I agree, let's keep host ports. The only stable pipeline of the otel collector currently is tracing. And running agent with no ports exposed doesn't provide any value for traces collection.

Reply:
> in my AWS EKS cluster, it fails to be deployed when trying to use hostPort. it seems that it requires additional cluster configuration to map it to host port. It would make it a bit more complicated to deploy it.

@rockb1017 Could you please share the errors you see? I never saw any issues with exposed ports on this deamonset.